src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/OnlineFormDocumentService.java

```diff
@@ -72,4 +73,42 @@ private void deleteOldMiniFormA(Map<String, Object> documentData, String authori
             }
         });
     }
-}
+
+    public CaseDocument generateConsentedInContestedMiniFormA(CaseDetails caseDetails, String authorisationToken) {
+
+        log.info("Generating 'Consented in Contested' Mini Form A for Case ID : {}", caseDetails.getId());
+
+        CaseDetails caseDetailsCopy = documentHelper.deepCopy(caseDetails, CaseDetails.class);
+        prepareMiniFormFields(caseDetailsCopy);
+
+        return genericDocumentService.generateDocument(authorisationToken, caseDetailsCopy,
+            documentConfiguration.getMiniFormTemplate(),
```

Comment:
would using the contested template avoid needing to add the fields below?

Reply:
Unfortunately the contested template has additional values that would be rendered, like fast track application values which does not exist in consented. A field mapping is required or a new document - this seemed like the most painless of the two